docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,567 @@
+2. Install [CMake][], [Ninja][] and [Sccache][]:
+   - Via [Homebrew][] (recommended):
+     ```sh
+     brew install cmake ninja sccache
+     ```
+   - Via [Homebrew Bundle][]:
+     ```sh
+     brew bundle
+     ```
+
+[Xcode]: https://developer.apple.com/xcode/resources/
+[CMake]: https://cmake.org
+[Ninja]: https://ninja-build.org
+[Sccache]: https://github.com/mozilla/sccache
+[Homebrew]: https://brew.sh/
+[Homebrew Bundle]: https://github.com/Homebrew/homebrew-bundle
+
+### Ubuntu Linux
+
+1. For Ubuntu 16.04 LTS and 18.04 LTS, run the following:
+
+   ```sh
+   sudo apt-get install    \
+     clang                 \
+     cmake                 \
+     git                   \
+     icu-devtools          \
+     libcurl4-openssl-dev  \
+     libedit-dev           \
+     libicu-dev            \
+     libncurses5-dev       \
+     libpython-dev         \
+     libsqlite3-dev        \
+     libxml2-dev           \
+     ninja-build           \
+     pkg-config            \
+     python                \
+     python-six            \
+     rsync                 \
+     swig                  \
+     systemtap-sdt-dev     \
+     tzdata                \
+     uuid-dev
+   sudo snap install sccache --candidate --classic
+   ```
+
+   For Ubuntu 20.04 LTS, you can run the same command
+   but with `libpython2-dev` instead of `libpython-dev`.
+
+   **Note:** LLDB currently requires at least `swig-1.3.40` but will
+   successfully build with version 2 shipped with Ubuntu.
+
+## Building the project for the first time
+
+### Spot check dependencies
+
+* Run `cmake --version`: This should be 3.18.1 or higher for macOS.
+* Run `python3 --version`: Check that this succeeds.
+* Run `ninja --version`: Check that this succeeds.
+* Run `sccache --version`: Check that this succeeds.
+
+### The roles of different tools
+
+At this point, it is worthwhile to pause for a moment
+to understand what the different tools do:
+
+1. CMake is a cross-platform build system for C and C++.
+   It forms the core infrastructure used to _configure_ builds of
+   Swift and its companion projects.
+2. Ninja is a low-level build system that can be used to _build_ the project,
+   as an alternative to Xcode's build system. Ninja is somewhat faster,
+   especially for incremental builds, and supports more build environments.
+3. Sccache is a caching tool:
+   If you ever delete your build directory and rebuild from scratch
+   (i.e. do a "clean build"), Sccache can accelerate the new build
+   significantly. There are few things more satisfying than seeing Sccache
+   cut through build times.
+4. `utils/update-checkout` is a script to help you work with all the individual
+   git repositories together, instead of manually cloning/updating each one.
+5. `utils/build-script` (we will introduce this shortly)
+   is a high-level automation script that handles configuration (via CMake),
+   building (via Ninja or Xcode), caching (via Sccache), running tests and more.
```

Comment:
It might be good to mention Xcode's role here too:

```markdown
1. On applicable platforms, the IDE (Xcode or Visual Studio) mainly serves as a package
   of development tools (like linkers and compilers) and SDKs (header files, etc.); the
   IDE itself and its build system aren't typically used when building Swift. On Linux,
   these tools and headers are usually installed through the distribution's package
   manager instead.
```